library/std/src/os/unix/fs.rs

```diff
@@ -985,6 +985,10 @@ impl DirBuilderExt for fs::DirBuilder {
 /// Changing the owner typically requires privileges, such as root or a specific capability.
 /// Changing the group typically requires either being the owner and a member of the group, or
 /// having privileges.
+/// 
+/// Be aware that changing owner clears the `suid` and `sgid` permission bits in most cases
+/// according to POSIX, usually even if the user is root. The sgid is not cleared when
+/// the file is non-group-executable.
```

Comment:
Absolutely. 
For example the [chown(2) manpage](https://www.man7.org/linux/man-pages/man2/chown.2.html) says:

> When the owner or group of an executable file is changed by an
       unprivileged user, the S_ISUID and S_ISGID mode bits are cleared.
       POSIX does not specify whether this also should happen when root
       does the chown(); the Linux behavior depends on the kernel
       version, and since Linux 2.2.13, root is treated like other
       users.  

> In case of a non-group-executable file (i.e., one for
       which the S_IXGRP bit is not set) the S_ISGID bit indicates
       mandatory locking, and is not cleared by a chown().

> When the owner or group of an executable file is changed (by any
       user), all capability sets for the file are cleared.

And as you properly quoted [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/chown.html) says:

> Unless chown is invoked by a process with appropriate privileges, the set-user-ID and set-group-ID bits of a regular file shall be cleared upon successful completion; the set-user-ID and set-group-ID bits of other file types may be cleared.

I have considered to quote any of those but that would've been excessive for this pretty and short help page, but linking any of them (especially the first) may be indeed useful.

Reply:
Yeah, I don't think we need (nor should) quote the *entire* thing, just having a link to the more detailed reference is helpful 👍